storage-operator/deploy/crds/storage_v1alpha1_volume_crd.yaml

```diff
@@ -67,6 +67,56 @@ spec:
           - storageClassName
           type: object
         status:
+          properties:
+            failedContext:
+              properties:
+                errorCode:
+                  enum:
+                  - 0
+                  - 1
+                  - 2
+                  - 3
+                  format: int64
+                  type: integer
+                errorMessage:
+                  type: string
+              type: object
+            pendingContext:
+              properties:
+                jobId:
+                  description: JID of the Salt job in progress.
+                  type: string
+                size:
+                  description: Size of the volume.
+                  type: string
+                step:
```

Comment:
I wish we would *not* need to store states/steps in the CR, but 'calculate' them from other information.

Reply:
But where would I store the JID between reschedule?
Or if I spawn a Salt job, I crash and restart, I need to have the JID somewhere to:

1. know that something is still in progress, no need to restart it
2. poll its state.

In the same vein, if I don't use "steps" and only store a `JobID` how can I know if I'm running `PrepareVolume` or `disk.dump`?

I tried to write a reconciliation loop that's resilient to crash and can resume from where it was upon restart, but that require to persist some kind of state somewhere (maybe the CR is not the good place though…).

Reply:
Done, I reworked the logic to calculate as much as possible the state instead of storing.